File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let size = [];
    for(let i=0; i<sizes.length; i++)
        size.push(sizes[i][0]>sizes[i][1]? sizes[i][1] : sizes[i][0]);
    return Math.max(...sizes.flat())*Math.max(...size.flat());
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.Arrays;
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int min[] = new int[sizes.length];
        int max[] = new int[sizes.length];
        
        for(int i=0; i<sizes.length; i++){
            min[i] = sizes[i][0] < sizes[i][1]? sizes[i][0] : sizes[i][1];
            max[i] = sizes[i][0] > sizes[i][1]? sizes[i][0] : sizes[i][1];
        }
        
        Arrays.sort(min);
        Arrays.sort(max);
        
        return min[sizes.length-1]*max[sizes.length-1];
    }
}